dace/codegen/targets/xilinx.py

```diff
@@ -193,19 +200,75 @@ def define_stream(dtype, vector_length, buffer_size, var_name, array_size,
             kernel_stream.write("dace::SetNames({}, \"{}\", {});".format(
                 var_name, var_name, cpp.sym2cpp(array_size)))
 
-    @staticmethod
-    def define_local_array(dtype, vector_length, var_name, array_size, storage,
-                           shape, function_stream, kernel_stream, sdfg,
-                           state_id, node):
+    def define_local_array(self, var_name, desc, array_size, veclen,
+                           function_stream, kernel_stream, sdfg, state_id,
+                           node):
         kernel_stream.write("dace::vec<{}, {}> {}[{}];\n".format(
-            dtype.ctype, vector_length, var_name, cpp.sym2cpp(array_size)))
-        if storage == dace.dtypes.StorageType.FPGA_Registers:
-            kernel_stream.write("#pragma HLS ARRAY_PARTITION variable={} "
-                                "complete\n".format(var_name))
-        elif len(shape) > 1:
-            kernel_stream.write("#pragma HLS ARRAY_PARTITION variable={} "
-                                "block factor={}\n".format(
-                                    var_name, shape[-2]))
+            desc.dtype.ctype, veclen, var_name, cpp.sym2cpp(array_size)))
+        if desc.storage == dace.dtypes.StorageType.FPGA_Registers:
+            kernel_stream.write(
+                "#pragma HLS ARRAY_PARTITION variable={} "
+                "complete\n".format(var_name), node, state_id, sdfg)
+        elif len(desc.shape) > 1:
```

Comment:
should this be more explicit w.r.t. storage types?

Reply:
Fixed